### ISSUE TITLE
Add CI check to ensure changes to combine API are compatible with other serivces

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#2853]
The main API/service now relies on the output of the combine service.  This is mediated by an automatically generated library (combine-api-client).

 A [script](../blob/dev/tools/generate-combine-client) automatically generates the library, including the typescript types and models for the responses based on the openAPI schema defined in the [repo](../blob/dev/apps/combine-service/spec/spec.yml). Then. the library is imported as needed in the other apps, and the typescript type checking can handle ensuring consistency. 

If any changes are made to the openAPI specification, then this library should be regenerated, and then the CI should be run to make sure the new library is still compatible with the rest of the repo. We should eventually add a CI script to ensure this is the case, but for now, we can regenerate the library if any changes are made to ensure everything stays in sync.

**Comment by jonrkarr** [comment]
This sounds good. 

What's the command to generate the library?

I tried to have the unit tests check that the output of the COMBINE API matches the API specs. I had to disable this once I needed to introduce schema components that the OpenAPI Python package couldn't handle (I guess it only supports a subset of OpenAPI). Ideally, we'd track this down and refactor the schema to work with all of the packages that we're using. For now, we have unit tests. Since they're not automated, there is slight risk that the same problem is replicated in the API implementation, tests, and client TypeScript code.

**Comment by bilalshaikh42** [comment]
> What's the command to generate the library?

just calling the 'tools/generate-combine-client' from the command line once #2850 is merged in

**Comment by jonrkarr** [comment]
Doing this well requires #2479. Unfortunately, I seemed to need to turn this off due to incompatibilities among apicur.io, connexion, and OpenAPI.

**Comment by bilalshaikh42** [comment]
Yup there are two levels of validation needed. 

1. The combine app output should match the schema 
2. The typescript codes handling of the combine  output should match the schema 

The method here to generate the client on changes to the schema will allow for 2. If the schema is not compatible with the handling, then the typescript  will have build errors. As you pointed out, verifying 1 is a separate issue that is also needed

**Comment by jonrkarr** [comment]
Using the new combine client would solve this. Two additional modules need to be mapped to the combine client.

`_type` attributes can be removed from the data models in `apps/dispatch/src/app` as needed.

- `apps/dispatch/src/app/services/combine/combine.service.ts`
  - `getSpecsOfSedDocsInCombineArchive` --> `SimulationProjectsCOMBINEOMEXArchivesService.srcHandlersCombineGetSedmlSpecsForCombineArchiveHandler`
  - `getCombineArchiveMetadata` --> `SimulationProjectsCOMBINEOMEXArchivesService.srcHandlersCombineGetMetadataForCombineArchiveHandlerBiosimulations`
  - `validateModel` --> `ModelService.srcHandlersModelValidateHandler`
  - `validateSimulation` --> `SimulationExperimentsSEDMLService.srcHandlersSedMlValidateHandler`
  - `validateMetadata` --> `MetadataOMEXMetadataService.srcHandlersOmexMetadataValidateHandler`
  - `validateProject` --> `SimulationProjectsCOMBINEOMEXArchivesService.srcHandlersCombineValidateHandler`
  - `getSimilarAlgorithms` --> `SimulationAlgorithmsKiSAOService.srcHandlersKisaoGetSimilarAlgorithmsHandler`
- `CreateSimulationProjectComponent`
  - `${urls.combineApi}sed-ml/get-parameters-variables-for-simulation` --> `SimulationExperimentsSEDMLService.srcHandlersSedMlGetParametersVariablesForSimulationHandler`
  - `${urls.combineApi}combine/create` --> `SimulationProjectsCOMBINEOMEXArchivesService.srcHandlersCombineCreateHandler`

**Comment by jonrkarr** [comment]
Done in #3350 

- 1 action re-builds COMBINE API and pushes it
- The main CI action will then test this and any other libraries/apps that depend on it

The action that builds the COMBINE API doesn't test it to avoid copying the testing workflow and then having a copy to maintain
